# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/ContentBlocking.cpp
# Commit: ae01b3db6250
# Full Hash: ae01b3db6250b366ac975c4257a384c1de13d632
# Author: Dimi Lee <dlee@mozilla.com>
# Date: 2020-04-23 21:43:09
# Regressor Bug: 1616775
# File Overlap Count: 1
# Description:
#   Bug 1616775 - P1. Add IsThirdPartyWindow and IsThirdPartyTrackingResourceWindow fields to WindowContext. r=timhuang,baku,farre
#   
#   We have to add "IsThirdPartyWindow" in WindowContext because we need to know if
#   a BrowsingContext is third-party (The browsing context may be not
#   in-process).
# ==============================================================================

diff -r 685b68485248 -r ae01b3db6250 toolkit/components/antitracking/ContentBlocking.cpp
--- a/toolkit/components/antitracking/ContentBlocking.cpp	Thu Apr 23 04:33:57 2020 +0000
+++ b/toolkit/components/antitracking/ContentBlocking.cpp	Thu Apr 23 14:24:56 2020 +0000
@@ -306,19 +306,15 @@
 
   } else {
     // We should be a 3rd party source.
-    // Make sure we are either a third-party tracker or a third-party
-    // window (depends upon the cookie bahavior).
     if (behavior == nsICookieService::BEHAVIOR_REJECT_TRACKER &&
-        !nsContentUtils::IsThirdPartyTrackingResourceWindow(
-            parentInnerWindow)) {
+        !parentWindowContext->GetIsThirdPartyTrackingResourceWindow()) {
       LOG(("Our window isn't a third-party tracking window"));
       return StorageAccessGrantPromise::CreateAndReject(false, __func__);
     } else if ((CookieJarSettings::IsRejectThirdPartyWithExceptions(behavior) ||
                 behavior ==
                     nsICookieService::
                         BEHAVIOR_REJECT_TRACKER_AND_PARTITION_FOREIGN) &&
-               !nsContentUtils::IsThirdPartyWindowOrChannel(parentInnerWindow,
-                                                            nullptr, nullptr)) {
+               !parentWindowContext->GetIsThirdPartyWindow()) {
       LOG(("Our window isn't a third-party window"));
       return StorageAccessGrantPromise::CreateAndReject(false, __func__);
     }
